### PR TITLE
Correlation ID

### PR DESCRIPTION
**Pull request by scarney81**
@apechimp @ruthlesshelp @lanetix/engineers - this pull request adds 3 methods to the request for microservices: `error`, `log`, `warn`. When employed, these methods also log the correlation ID with the message.

This pull request also modifies the error middleware to make use of the `error` function that is appended to the request object.

eslint cleanup was also required as there were breaking items :sob:

### CODE REVIEW

File: middleware/error.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = function (app) {
         // for Papertrail purposes.
         var errorMsg = 'Error: ' + (err.stack || util.inspect(err, { depth: null }));
         errorMsg += req ? '\nUrl: ' + req.originalUrl + '\nBody: ' + util.inspect(req.body, { depth: null }) : '';
-        console.error(errorMsg);
+        req.error(errorMsg);
```
Comment (scarney81) [inline middleware/error.js]: Use new error function

File: middleware/jwt.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ module.exports = function (app) {
             }
           });
         } else {
-          console.error('Error verifying JWT: ', err);
+          req.error('Error verifying JWT: ', err);
```
Comment (scarney81) [inline middleware/jwt.js]: Use new error function

File: middleware/log.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+'use strict';
```
Comment (scarney81) [inline middleware/log.js]: This is the code that appends the functions

File: server.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ module.exports = function (app, options) {
   if (options.configure && _.isFunction(options.configure)) {
     options.configure(app);
   }
+  app.use(middleware.log);
```
Comment (scarney81) [inline server.js]: Ensure that it's the first thing appended.

File: .eslintrc [hunk 5]
```diff
@@ -16,7 +16,6 @@
     "quotes": [2, "single", "avoid-escape"],
     "no-undef": 2,
     "no-unused-vars": 2,
-    "strict": 2,
```
Comment (scarney81) [inline .eslintrc]: This was a duplicate line in eslint config

**Comment by sahiltalwar88** [comment]
🚍

File: server.js [hunk 6]
```diff
@@ -8,8 +8,8 @@ var _ = require('lodash'),
   middleware = require('require-directory')(module, path.join(__dirname, 'middleware'));
 
 module.exports = function (app, options) {
-  _.extend(options, config)
-  options = _.extend(config, options)
+  _.extend(options, config);
+  options = _.extend(config, options);
```
Comment (jcready) [inline server.js]: These two lines are... confusing.

Comment (scarney81) [inline server.js]: @jcready - they are. Good thing I was only adding a semi-colon to pass lint here :trollface:

**Pull request merged**